### ISSUE TITLE
"best match" metric for mason search

### ISSUE THREAD

**Issue by ankingcodes** [chapel-lang/chapel#14968]
Currently, `mason search` simply displays the list of packages without any ranking system. This would be a bad design in the future, when Chapel users make multiple packages with names that are similar to each other or provide similar functionality. Therefore, we could have a "best match" metric which would rank the packages corresponding to a `mason search`. 
Some constraints were listed in #9846 which were meant for CI testing. The very same constraints could be used to map a package to a rank.
For example, `npm` ranks its packages on the basis of popularity, quality, maintenance. We could do the same for Mason but only on the basis of quality ( presence of README, stability, tests, examples) and maintenance(checking the version number).

**Comment by ben-albrecht** [comment]
Great idea @ankingcodes.

Here are some design questions to consider:

> presence of README, stability, tests, examples) and maintenance(checking the version number).

What is the actual metric to be used? Do each of these checkboxes count as a point, or are they weighted in some way?

One challenge will be how to store and update the metric data automatically. The registry only contains the TOML manifest files, so we will have to embed the metrics about the package contents somewhere in the registry.

Should there be a flag to revert back to alphabetical sorting? (I would think yes)

**Comment by ankingcodes** [comment]
> What is the actual metric to be used? Do each of these checkboxes count as a point, or are they weighted in some way?

@ben-albrecht Initially, I believe we should give a point for each of the checkboxes. Since, we don't have to assess packages on the basis of popularity right now. 

> One challenge will be how to store and update the metric data automatically. 

Regarding storing the metric data, I believe we could create another file within the registry which contains the metrics along with the package_name. The `mason search` command could benefit from this too, making it even faster. 

> Should there be a flag to revert back to alphabetical sorting? (I would think yes)

Absolutely !

**Comment by ben-albrecht** [comment]
> Do you have any ideas to automate this ? I think this can be taken care during CI.

Yes, I think CI could handle making modifications to the file post-merge. This would prevent any potential merge conflicts as well. Perhaps we could use a TOML format for this file (for consistency-sake), and store
